# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/WebCodecsUtils.h
# Commit: d7f55df2c2e4
# Full Hash: d7f55df2c2e4305928afec2b632f72517ccfb35f
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-29 15:31:45
# Regressor Bug: 1831451
# File Overlap Count: 2
# Description:
#   Bug 1831451 - Guess color space from VideoData r=aosmond
#   
#   Depends on D171940
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D170635
# ==============================================================================

diff -r 61be9a6b9cd4 -r d7f55df2c2e4 dom/media/webcodecs/WebCodecsUtils.h
--- a/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 04:15:03 2023 +0000
+++ b/dom/media/webcodecs/WebCodecsUtils.h	Thu Jun 29 04:15:03 2023 +0000
@@ -11,6 +11,7 @@
 
 #include "ErrorList.h"
 #include "js/TypeDecls.h"
+#include "mozilla/Maybe.h"
 #include "mozilla/Result.h"
 #include "mozilla/Span.h"
 #include "mozilla/dom/UnionTypes.h"
@@ -68,6 +69,16 @@
 
 gfx::ColorSpace2 ToPrimaries(VideoColorPrimaries aPrimaries);
 
+bool ToFullRange(const gfx::ColorRange& aColorRange);
+
+Maybe<VideoMatrixCoefficients> ToMatrixCoefficients(
+    const gfx::YUVColorSpace& aColorSpace);
+
+Maybe<VideoTransferCharacteristics> ToTransferCharacteristics(
+    const gfx::TransferFunction& aTransferFunction);
+
+Maybe<VideoColorPrimaries> ToPrimaries(const gfx::ColorSpace2& aColorSpace);
+
 }  // namespace dom
 }  // namespace mozilla
 